catalog/solr/catalog-solr-offline-gazetteer/src/main/java/ddf/catalog/solr/offlinegazetteer/RemoveAllOfflineSolrGazetteerCommand.java

```diff
@@ -0,0 +1,91 @@
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.solr.offlinegazetteer;
+
+import com.google.common.collect.ImmutableList;
+import java.util.concurrent.TimeUnit;
+import net.jodah.failsafe.Failsafe;
+import net.jodah.failsafe.RetryPolicy;
+import org.apache.karaf.shell.api.action.Action;
+import org.apache.karaf.shell.api.action.Command;
+import org.apache.karaf.shell.api.action.Option;
+import org.apache.karaf.shell.api.action.lifecycle.Reference;
+import org.apache.karaf.shell.api.action.lifecycle.Service;
+import org.apache.karaf.shell.api.console.Session;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.codice.solr.client.solrj.SolrClient;
+import org.codice.solr.client.solrj.UnavailableSolrException;
+import org.codice.solr.factory.SolrClientFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Service
+@Command(
+  scope = "offline-solr-gazetteer",
+  name = "removeall",
+  description = "Deletes all items in the offline solr gazetteer solr core"
+)
+public class RemoveAllOfflineSolrGazetteerCommand implements Action {
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(RemoveAllOfflineSolrGazetteerCommand.class);
+
+  @Reference protected Session session;
+
+  @Reference private SolrClientFactory clientFactory;
+
+  @Option(
+    name = "--force",
+    aliases = {"-f"},
+    description = "Force the removal without a confirmation message."
+  )
+  boolean force = false;
+
+  private final RetryPolicy retryPolicy =
+      new RetryPolicy()
+          .retryOn(ImmutableList.of(UnavailableSolrException.class, SolrServerException.class))
+          .withMaxDuration(5, TimeUnit.SECONDS)
+          .withBackoff(25, 1_000, TimeUnit.MILLISECONDS);
+
+  @Override
+  public Object execute() throws Exception {
+    if (!force) {
+      String answer =
+          session
+              .readLine(
+                  "Are you sure you want to remove all gazetteer entries inside of the offline solr gazetteer core?(y/n)",
+                  ' ')
+              .toLowerCase();
+      if (!("y".equals(answer) || "yes".equals(answer))) {
+        session.getConsole().println("Aborting.");
+        return null;
+      }
+    }
+    try {
+      SolrClient solrClient =
+          clientFactory.newClient(OfflineGazetteerPlugin.STANDALONE_GAZETTEER_CORE_NAME);
+
+      Failsafe.with(retryPolicy).get(() -> solrClient.deleteByQuery("*:*"));
+    } catch (Exception e) {
+      LOGGER.info("Error while executing", e);
+      session.getConsole().println("Error while submitting remove all, exiting.");
+      throw e;
+    }
+    session.getConsole().println("Removeall submitted successfully.");
+    return null;
+  }
+
+  public void setClientFactory(SolrClientFactory clientFactory) {
```

Comment:
❓ I don't remember if Karaf supports constructor injection for commands. I think there are issues with it. I know we have tried to do it in the past but hit issues and had to resort to bundle context manipulation: 
https://github.com/codice/ddf-ui/blob/master/ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/commands/SearchFormsManageCommand.java#L77
https://github.com/codice/ddf-ui/blob/master/ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/commands/SearchFormsLoaderCommand.java#L75

That being said do we need this setter?

Reply:
Don't think we need the setter 👍